src/module.c

```
@@ -7815,9 +7815,8 @@ ValkeyModuleBlockedClient *moduleBlockClient(ValkeyModuleCtx *ctx,
                                              void *privdata,
                                              int flags) {
     client *c = ctx->client;
-    if (c->flag.blocked || getClientType(c) != CLIENT_TYPE_NORMAL || c->flag.deny_blocking) {
-        /* Early return if duplicate block attempt or client is not normal or
-         * client is set to deny blocking. */
+    if (c->flag.blocked || getClientType(c) != CLIENT_TYPE_NORMAL) {
```

I don't understand why we can delete deny_blocking here. Is the client type always != NORMAL for deny_blocking clients?

---

I think @enjoy-binbin is assuming that whenever we get to this code with `deny_blocking == 1` it means we are either running in the context of a script or a MULTI command. And in that case, we only want to prevent blocking the client if `is_keyspace_notification` is true

---

yes, if we keep this deny_blocking condition here, there will be some dead code like this one. If we are from lua or multi, we always set deny_blocking so this code become a dead code. In this case, i think we should return the correct error (or at least at that time i think, i did kind of forget something around it.)
```
    if (islua || ismulti) {
        c->bstate->module_blocked_handle = NULL;
        addReplyError(c, islua ? "Blocking module command called from Lua script"
```

---

Can we add an assert before this line to ensure the assumptions are correct?

```c
serverAssert(!c->flag.deny_blocking || (islua || ismulti));
```

---

Thanks! Added